superset-frontend/src/SqlLab/components/ResultSet/index.tsx

```diff
@@ -250,18 +275,7 @@ export default class ResultSet extends React.PureComponent<
               this.props.database?.allows_virtual_table_explore && (
                 <ExploreResultsButton
                   database={this.props.database}
-                  onClick={() => {
-                    // There is currently redux / state issue where sometimes a query will have serverId
-                    // and other times it will not.  We need this attribute consistently for this to work
-                    // const qid = this.props?.query?.results?.query_id;
-                    // if (qid) {
-                    //   // This will open explore using the query as datasource
-                    //   window.location.href = `/explore/?dataset_type=query&dataset_id=${qid}`;
-                    // } else {
-                    //   this.setState({ showSaveDatasetModal: true });
-                    // }
-                    this.setState({ showSaveDatasetModal: true });
-                  }}
+                  onClick={this.createExploreResultsOnClick}
```

Comment:
we still want to hide this feature so leave the original code in place, and add a comment saying to enable chart source to query change the onClick to `onClick={this.createExploreResultsOnClick}`

Reply:
yeah that makes sense, just did that, though will reverse it after this has gone through code review so that QA can test the new flow.